pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1555,261 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
 
 	response.WriteHeader(http.StatusAccepted)
 }
+
+func generateVMIInterfaceRequestPatch(vmi *v1.VirtualMachineInstance, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if alreadyExists(interfaceRequest, iface) {
+			return "", fmt.Errorf("unable to add interface [%s] because it already exists", iface.Name)
+		}
+	}
+
+	vmiCopy := vmi.DeepCopy()
+	vmiCopy.Spec = *ApplyInterfaceRequestOnVMISpec(&vmiCopy.Spec, interfaceRequest)
+
+	oldIfacesJSON, err := json.Marshal(vmi.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	newIfacesJSON, err := json.Marshal(vmiCopy.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	oldNetworksJSON, err := json.Marshal(vmi.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	newNetworksJSON, err := json.Marshal(vmiCopy.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	testNetworks := fmt.Sprintf(`{ "op": "test", "path": "/spec/networks", "value": %s}`, string(oldNetworksJSON))
+	updateNetworks := fmt.Sprintf(`{ "op": "%s", "path": "/spec/networks", "value": %s}`, "replace", string(newNetworksJSON))
+
+	testInterfaces := fmt.Sprintf(`{ "op": "test", "path": "/spec/domain/devices/interfaces", "value": %s}`, string(oldIfacesJSON))
+	updateInterfaces := fmt.Sprintf(`{ "op": "%s", "path": "/spec/domain/devices/interfaces", "value": %s}`, "replace", string(newIfacesJSON))
+
+	patch := fmt.Sprintf("[%s, %s, %s, %s]", testNetworks, testInterfaces, updateNetworks, updateInterfaces)
+	return patch, nil
+}
+
+func alreadyExists(interfaceRequest *v1.VirtualMachineInterfaceRequest, iface v1.Interface) bool {
+	return interfaceRequest.AddInterfaceOptions != nil && iface.Name == dynamicIfaceName(interfaceRequest)
+}
+
+func generateVMInterfaceRequestPatch(vm *v1.VirtualMachine, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
```

Comment:
I think I understand the VMI patching, where the interface and network entries are added to the spec.
I thought the same will occur for the VM spec as well, but it seems that for the VM, we pass through a different subtree under the `status`section. Something that is not that trivial to me.
(just because usually requests are some desired state, which should appear on the spec and not on the status)

This raises the question (for me), who is patching the VM spec template section, where the interfaces and networks are defined.

Reply:
because this is the kubevirt pattern on how to reconcile the VM / VMI objects.

If we followed your suggestion (multiple people said the same, so you're not alone ...) the client would have to patch the VMI spec and the VM spec, which would risk getting the VM / VMI into a limbo state - let's assume one of the calls succeeds and the other fails). We want to prevent that from ever happening.

More details in [this slack thread](https://kubernetes.slack.com/archives/C0163DT0R8X/p1674662747056389?thread_ts=1674570833.329709&cid=C0163DT0R8X).